app/js/components/homePage/manageApps/manageApps.jsx

```diff
@@ -7,22 +7,57 @@
  * graphic logo is a trademark of OpenMRS Inc.
  */
 import React from 'react';
+import fetchPolyfill from 'whatwg-fetch';
+import AddAddon from '../manageApps/AddAddon.jsx';
+import { ApiHelper } from '../../../helpers/apiHelper';
 
 export default class ManageApps extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.handleClear = this.handleClear.bind(this);
+    this.handleUpload = this.handleUpload.bind(this);  
+  }
+  
+  componentDidMount() {
+    $(":file").filestyle({btnClass: "btn-primary"});
+  }
+
+  handleUpload() {
+    const applicationDistribution = location.href.split('/')[2];
+    const url = location.href.split('/')[3];
+    let apiBaseUrl = `/${applicationDistribution}/${url}/ws/rest`;
+    this.requestUrl ='/owa/addapp';
+
+    const fd = new FormData();
+    fd.append('file', document.getElementById('fileInput').files[0]);
+    $.ajax({
```

Comment:
@andela-jomadoye You probably will have to declare $ as a global variable and define it to require query, else tests will fail 😄

Reply:
@paulupendo Test is passing.